include/zephyr/drivers/video.h

```diff
@@ -202,6 +202,33 @@ struct video_frmival_enum {
 	};
 };
 
+/**
+ * @brief Port / Endpoint DT Helpers
+ *
+ */
+
+/* Drivers should not use this except they know exactly that port has a pid */
+#define _DT_INST_PORT_BY_ID(n, pid)                                                                \
+	COND_CODE_1(DT_NODE_EXISTS(DT_INST_CHILD(n, ports)), (DT_CHILD(DT_INST_CHILD(n, ports), port_##pid)), (DT_INST_CHILD(n, port_##pid)))
+
+/* Drivers rarely use this as they usually deals with only endpoint */
+#define DT_INST_PORT_BY_ID(n, pid)                                                                 \
+	COND_CODE_1(DT_NODE_EXISTS(_DT_INST_PORT_BY_ID(n, pid)), (_DT_INST_PORT_BY_ID(n, pid)), (DT_INST_CHILD(n, port)))
+
+/* Drivers should not use this except they know exactly that the endpoint has a pid */
+#define _DT_INST_ENDPOINT_BY_ID(n, pid, id) DT_CHILD(DT_INST_PORT_BY_ID(n, pid), endpoint_##id)
+
+#define DT_INST_ENDPOINT_BY_ID(n, pid, id)                                                         \
+	COND_CODE_1(DT_NODE_EXISTS(_DT_INST_ENDPOINT_BY_ID(n, pid, id)), (_DT_INST_ENDPOINT_BY_ID(n, pid, id)), (DT_CHILD(DT_INST_PORT_BY_ID(n, pid), endpoint)))
+
+/* Drivers should not use this except they know exactly that the remote device has no ports node */
+#define DT_REMOTE_DEVICE_NO_PORTS(ep)                                                              \
```

Comment:
Question here- it seems like in Zephyr we only currently support this scheme for remote ports:
```
	mt9m114: mt9m114@48 {
		compatible = "aptina,mt9m114";
		reg = <0x48>;

		port {
			mt9m114_ep_out: endpoint {
				remote-endpoint-label = "dfi_ep_in";
			};
		};
	};
```

So do we need to supply this macro yet?

Reply:
Ok, I see how we are doing this with multiple child bindings now- sorry I missed that.